app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/generate/GenerateTravelCertificateViewModel.kt

```diff
@@ -23,30 +19,31 @@ import kotlinx.datetime.TimeZone
 import kotlinx.datetime.toLocalDateTime
 
 internal class GenerateTravelCertificateViewModel(
+  contractId: String?,
   private val getTravelCertificateSpecificationsUseCase: GetTravelCertificateSpecificationsUseCase,
   private val createTravelCertificateUseCase: CreateTravelCertificateUseCase,
-  private val downloadTravelCertificateUseCase: DownloadTravelCertificateUseCase,
 ) : ViewModel() {
   private val _uiState: MutableStateFlow<TravelCertificateInputState> = MutableStateFlow(TravelCertificateInputState())
   val uiState: StateFlow<TravelCertificateInputState> = _uiState.asStateFlow()
 
   init {
-    updateInfo()
+    updateSpecifications(contractId)
   }
 
-  private fun updateInfo() {
+  private fun updateSpecifications(contractId: String?) {
     viewModelScope.launch {
       _uiState.update { it.copy(isLoading = true) }
       getTravelCertificateSpecificationsUseCase
-        .invoke()
+        .invoke(contractId)
         .fold(
           ifLeft = { travelCertificateError ->
             _uiState.update {
               when (travelCertificateError) {
+                // todo: probably should not only show error dialog, but also don't show the input ui at all,
+                // todo: otherwise you have an opportunity to fill email etc manually, press continue and crash the app
```

Comment:
Yes, let's try and keep this in mind when we eventually work a bit more on cleaning up this feature.